### PR TITLE
Attempt to support various underscore.js imports

### PR DESCRIPTION
**Pull request by caseycesari**
## Overview
Adjust the use config to support all of the various ways that underscore
is imported within plugins.

The previous fix implemented in #1044 caused problems with plugins that
bring in their own version of underscore in via a CDN. The adjustments
here should make that possible while still allowing plugins to import
underscore via Use.js.

Also, rename the framework version of underscore to reduce confusion.

Connects to #669, #1043, #1046

### Notes

I'm not totally sure why this is working. It was arrived at through various tweaks of the Use config until various plugins that use underscore were all working.

## Testing Instructions

- Bring in regional-planning, flood-slr, and living-shorelines-explorer, and any other plugins that you know of that use underscore.
- Launch the framework, and ensure all of the plugins are working and that there are no errors related to underscore in the console (either errors finding underscore or undefined underscore methods.).
- Ensure regional-planning works before and after https://github.com/CoastalResilienceNetwork/regional-planning/pull/95. This will exercise two different approaches to using underscore.

### CODE REVIEW

File: src/GeositeFramework/Views/Home/Index.cshtml [hunk 1]
```diff
@@ -815,7 +815,8 @@
         useConfig['jqueryui'] = { attach: 'jqueryui' };
         useConfig['chosen'] = { attach: 'chosen', deps: ['jquery'] };
         useConfig['pips'] = { attach: 'pips', deps: ['jqueryui'] };
-        useConfig['underscore'] = { attach: 'underscore' };
+        useConfig['underscore'] = { attach: '_', deps: ['underscore'] };
+        useConfig['underscore-min'] = { attach: '_', deps: ['underscore'] };
```
Comment (caseycesari) [inline src/GeositeFramework/Views/Home/Index.cshtml]: Not sure why `underscore` has to be a dependency of `underscore` here, but if you don't do this underscore is defined as `3` in some plugins.

File: src/GeositeFramework/Views/Home/Index.cshtml [hunk 2]
```diff
@@ -843,7 +844,9 @@
                 "jqueryui": location.pathname.replace(/\/[^/]+$/, "") + "js/lib/jquery-ui.min",
                 "chosen": location.pathname.replace(/\/[^/]+$/, "") + "js/lib/chosen.jquery.min",
                 "pips": location.pathname.replace(/\/[^/]+$/, "") + "js/lib/jquery-ui-slider-pips.min",
-                "underscore": location.pathname.replace(/\/[^/]+$/, "") + "js/lib/underscore-min"
+                "underscore": location.pathname.replace(/\/[^/]+$/, "") + "js/lib/underscore",
+                "underscore/underscore-min": location.pathname.replace(/\/[^/]+$/, "") + "js/lib/underscore",
+                "underscore/underscore": location.pathname.replace(/\/[^/]+$/, "") + "js/lib/underscore",
```
Comment (caseycesari) [inline src/GeositeFramework/Views/Home/Index.cshtml]: I think this is capturing cases where underscore is brought in via a CDN. I'm attempting to redirect them to the framework underscore.

**Comment by kellyi** [comment]
Set up these plugins to test:

```sh
ecological-effects-slc/
flood-slr/
future_habitat_v2/
launchpad/
living-shorelines-explorer/
natural-coastal-protection/
recreation_tourism/
regional-planning/
subregion_toggle/
tuna-fisheries/
```

**Comment by kellyi** [comment]
Tried it out with those plugins and verified that 1. they won't work on `develop` due to the underscore issue and 2. they do load on this branch:

screen shot 2018-01-04 at 11 52 25 am

I don't know whether this is known, prexisting, or newly-created problem but it looks like future habitat can't find `pips`:

screen shot 2018-01-04 at 11 54 18 am

**Comment by kellyi** [comment]
(That was for the `master` branches of those plugins, btw; going to check the `development` branches now).

**Comment by kellyi** [comment]
This seems to work in almost all cases. I was able periodically to reproduce the bug with Living Shorelines in Firefox with the following steps:

- have Living Shorelines, ENBA, and Launchpad as the installed plugins
- load the app in Firefox and by or before the time the launchpad screen opens, click on the Living Shorelines plugin to open it.

I'd see lines related to underscore errors in the console:

screen shot 2018-01-04 at 12 59 06 pm

It did take some work to cause that to happen though, but it makes me wonder whether it's an underlying timing issue -- and might pop up in slower browsers?

**Comment by caseycesari** [comment]
Thanks for the thorough reviews.

**Pull request merged**